### PR TITLE
Use shared constants for cache key prefixes

### PR DESCRIPTION
**Pull request by bobrenjc93**
## Summary
Centralize the shared cache key namespace prefixes in `torch/_inductor/codecache.py` and reuse them from the AOTAutograd cache key builder.

## Root cause
The `"c"` and `"a"` cache key namespace prefixes were hardcoded in separate modules. That duplicated cache infrastructure details across files and made it easy for related cache key builders to drift.

## Proposed fix
Define `FXGRAPH_CACHE_PREFIX = "c"` and `AOTAUTOGRAD_CACHE_PREFIX = "a"` in `torch/_inductor/codecache.py`, use `FXGRAPH_CACHE_PREFIX` in `code_hash()`, and import `AOTAUTOGRAD_CACHE_PREFIX` into `autograd_cache.py` for AOTAutograd key construction.

## Why this is the right long term fix
`codecache.py` owns the base cache infrastructure, so keeping the namespace prefixes there gives them a single source of truth and reduces the chance of future cache key inconsistencies.

## Testing
- `python3 -m py_compile torch/_inductor/codecache.py torch/_functorch/_aot_autograd/autograd_cache.py`
- AST validation script checking the new constants and both updated key constructors
- `rg` verification that the old bare `"a"`/`"c"` cache key prefix literals are gone from the touched constructors

Drafted via Codex, published after manual review by @bobrenjc93

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by bobrenjc93** [comment]
@claude

**Comment by bobrenjc93** [comment]
Addressed the follow-up feedback from https://github.com/pytorch/pytorch/pull/180093#issuecomment-4227549333 by centralizing the remaining `"f"` namespace prefix as `COMPILED_FX_GRAPH_CACHE_PREFIX` in `torch/_inductor/codecache.py`, alongside the existing shared `"c"` and `"a"` constants.

Local validation for this follow-up was:
- `python3 -m py_compile torch/_inductor/codecache.py torch/_functorch/_aot_autograd/autograd_cache.py`
- a targeted AST check that `code_hash()`, `compiled_fx_graph_hash()`, and `autograd_cache_key()` all build keys from the shared constants

**Comment by bobrenjc93** [comment]
@claude

**Comment by bobrenjc93** [comment]
Addressed three deterministic failures from the previous CI run:

- `torch/testing/_internal/custom_op_db.py`: detach tensors before calling `numpy()` so opcheck sample inputs with `requires_grad=True` do not fail schema checks.
- `aten/tools/run_tests.sh`: preserve `LD_LIBRARY_PATH` when invoking the ATen valgrind binaries directly so `basic` and `tensor_interop_test` can load `libtorch.so`.
- JIT async C++ tests now resolve `test_interpreter_async.pt` from the repo root at runtime instead of from the compile-time `__FILE__` checkout path.

Local validation on this host was limited to source-level checks because this shell does not have a torch-enabled Python environment or Python packaging toolchain available:
- `python3 -m py_compile torch/testing/_internal/custom_op_db.py`
- `bash -n aten/tools/run_tests.sh`
- `git diff --check`

**Comment by bobrenjc93** [comment]
@claude

**Comment by bobrenjc93** [comment]
@claude

**Comment by bobrenjc93** [comment]
Addressed the current CI build regression on the branch:

- `test/cpp/jit/test_utils.h`: replace nonexistent `#include <c10/util/fs.h>` with `#include <c10/util/FileSystem.h>`, which is the header that actually defines `c10::filesystem` in this repo.

Local validation on this host:
- `git diff --check`
- `g++ -std=gnu++20 -I/tmp/repos/pytorch/pytorch -c /tmp/fs_smoke.cpp -o /tmp/fs_smoke.o` against `c10/util/FileSystem.h`
- a best-effort `g++ -fsyntax-only` smoke compile of a TU including `test/cpp/jit/test_utils.h`, which now gets past the old missing `c10/util/fs.h` failure and stops later on missing generated build headers (`torch/headeronly/macros/cmake_macros.h`) because this checkout does not have a configured local PyTorch build tree

**Comment by bobrenjc93** [comment]
@claude

**Comment by bobrenjc93** [comment]
Removed the unrelated CI-fix commits from this branch after checking `git show`; the PR diff is back to the two cache-prefix files only.

@claude

**Review by aorenste — APPROVED** [review]
